src/Donors/Models/Donor.php

```diff
@@ -0,0 +1,227 @@
+        'createdAt' => DateTime::class,
+        'name' => 'string',
+        'firstName' => 'string',
+        'lastName' => 'string',
+        'email' => 'string',
+        'prefix' => 'string',
+    ];
+
+    /**
+     * @unreleased
+     *
+     * @param $id
+     *
+     * @return Donor
+     */
+    public static function find($id)
+    {
+        return give()->donorRepository->getById($id);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string  $email
+     * @return Donor
+     */
+    public static function whereEmail($email)
+    {
+        return give()->donorRepository->getByEmail($email);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string  $email
+     * @return bool
+     */
+    public function hasEmail($email)
+    {
+        $donor = give()->donorRepository->getByEmail($email);
+
+        return !is_null($donor);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  int  $userId
+     * @return Donor
+     */
+    public static function whereUserId($userId)
+    {
+        return give()->donorRepository->getByWpUserId($userId);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string  $email
+     * @return bool
+     * @throws Exception
+     */
+    public function addAdditionalEmail($email)
+    {
+        return give()->donorRepository->insertAdditionalEmail($this->id, $email);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  array  $attributes
+     *
+     * @return Donor
+     *
+     * @throws Exception|InvalidArgumentException
+     */
+    public static function create(array $attributes)
+    {
+        $donor = new static($attributes);
+
+        return give()->donorRepository->insert($donor);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return Donor
+     *
+     * @throws Exception|InvalidArgumentException
+     */
+    public function save()
+    {
+        if (!$this->id) {
+            return give()->donorRepository->insert($this);
+        }
+
+        return give()->donorRepository->update($this);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function delete()
+    {
+        return give()->donorRepository->delete($this);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return QueryBuilder
+     */
+    public function donations()
+    {
+        return give()->donations->queryByDonorId($this->id);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return QueryBuilder
+     */
+    public function subscriptions()
+    {
+        return give()->subscriptions->queryByDonorId($this->id);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return array
+     */
+    public function additionalEmails()
+    {
+        return give()->donorRepository->getAdditionalEmails($this->id);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return int
+     */
+    public function totalDonations()
+    {
+        return (int)give()->donations->getTotalDonationCountByDonorId($this->id);
```

Comment:
Doesn't seem like this needs to be cast again?

Reply:
I moved the cast into the repository 297d04c5

Reply:
We should update it within the QB, then. Wouldn't it make sense that count always returns an integer?